lib/rules/forbid-foreign-prop-types.js

```diff
@@ -34,26 +48,58 @@ module.exports = {
       return node.parent.type === 'AssignmentExpression' && node.parent.left === node;
     }
 
+    function isInPropTypesAssignment(node) {
+      if (!allowInPropTypes) {
+        return false;
+      }
+
+      if (node.parent.parent.parent.parent.type === 'AssignmentExpression') {
```

Comment:
are we sure that the node will always have 4 parents?

Reply:
Also, rather than checking for AssignmentExpressions, is there a way we could create a visitor for AssignmentExpression instead?

Reply:
No, I'm not sure the node will always have 4 parents. I'm sure there's a better way to handle this but I don't know what it is. Is there a method I can use to search the tree for a parent that's an `AssignmentExpression`? 

Creating a visitor for `AssignmentExpression` might be better but I'm not sure how to coordinate between the two visitors. If we find an `AssignmentExpression` where `x.propTypes` is on the left and `allowInPropTypes` is true then we need to ignore the rest of that block of code but the `MemberExpression` visitor will still be called for expressions inside that block of code. I'm not sure how to solve that.

Reply:
Good question.

Maybe we should make a shared utility function `isInAssignmentExpression` that walks upwards until it finds an `AssignmentExpression`?

Reply:
On the one hand I'm glad I didn't miss something obvious but on the other hand I wish there was an easy solution to this problem. 😀 

I'll try adding a function that traverses the tree upwards until it reaches an `AssignmentExpression`. What are some other node types that should make it stop? I'm thinking `Program` and `FunctionExpression` at the very least. Anything else?

Reply:
Program, certainly. Not so sure about FunctionExpression.